icu4c/source/common/cmemory.h

```diff
@@ -771,6 +783,19 @@ class MaybeStackVector : protected MemoryPool<T, stackCapacity> {
         return this->create(args...);
     }
 
+    template <typename... Args>
+    T *emplaceBackAndCheckErrorCode(UErrorCode &status, Args &&... args) {
+        if (U_FAILURE(status)) {
```

Comment:
Optional: Down here you can probably get away with

```
return this->createAndCheckErrorCode(status, args...);
```

Reply:
Done. I intend to squash into the first commit, leaving it attributed to @younies

Reply:
I think you changed it in the wrong place.

Is there a unit test dedicated to these classes? If there is, you should add a unit test.

Reply:
Oh great. Partial attention bites again. :-( Sorry!

There are no direct tests for MemoryPool or MaybeStackVector yet. What I had failed to do was merge into the PR which uses it and run that.

Reply:
I see the trie tests are also under test/intltest/ - so it seems the right location for tests for source/common/.

Reply:
(I wonder why it is called "cmemory" anyway, when it's C++ code?)